src/core/strategies/trigger.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2023 Open Assessment Technologies SA ;
+ */
+
+import { counterFactory } from '../../utils';
+import { isFunctionOperator } from '../terms';
+import tokensHelper from '../tokens.js';
+
+/**
+ * List of known strategies to apply to the current tokens when adding a new term.
+ * This will help to decide if existing tokens need to be evaluated prior to add the term.
+ * Each strategy expect a list of tokens in this order: ..., current, next.
+ * The result will be `true` if the current tokens must be evaluated.
+ * Otherwise, it will be `null` if no strategy applies.
+ * @type {contextStrategy[]}
+ */
+export const triggerStrategies = [
+    /**
+     * Checks if the expression contains enough terms to let the next strategy applies.
+     * @param {token[]} tokens - The list of tokens on which apply the strategy.
+     * @returns {boolean|null} - Returns `false` if the expression does not contain enough tokens.
+     * Otherwise, returns `null` if the next strategy could apply.
+     */
+    function expressionFilled(tokens = []) {
+        if (tokens.length < 4) {
```

Comment:
Are you sure about that, what about expression like `3!+` - shouldn't factorial be evaluated in that case first?

Reply:
Yes, the minimal expression for the strategy to apply is `operand operator operand`, and the minimal size is 3 chars + the new possible operator. The minimal expression with this respect is something like `1+2`, with an operator it may be `1+2+`.
The trigger applies only to binary operators. Unary operators are considered part of their operand and are discarded from the instant computation. Otherwise, it will complexity the process, because we would also need to consider cases like `1+3!` where the factorial would need to be processed as soon as the `!` operator is inserted.

Reply:
Thanks for explanations, now it's clear.